platform/android/MapboxGLAndroidSDK/src/test/java/com/mapbox/mapboxsdk/style/expressions/ExpressionTest.java

```diff
@@ -1272,4 +1275,19 @@ public void testRawMatchNumbers() {
       stop("layer2", 2.7));
     assertEquals("expressions should match", expected, raw(expected.toString()));
   }
+
+  @Test
+  public void testAlphaValueInColorConversion() {
+    // regression test for #12198
+    Expression colorExpression = color(Color.parseColor("#40FF0000")); // 25% alpha red
+    Object[] result = colorExpression.toArray();
+    assertEquals("alpha value should match", 0.25, (Float) result[4], 0.01f);
+  }
+
+  @Test
+  public void testAlphaValueInStringConversion() {
+    String color = PropertyFactory.colorToRgbaString(Color.parseColor("#80FF0000")); // 50% alpha red
+    String alpha = color.split(" ")[3].substring(0, 3);
```

Comment:
This test case doesn't really test the implementation as we are cutting off the number on the first decimal point.